src/TNodeImp.h

```diff
@@ -0,0 +1,80 @@
+//
+// Created by shvmo on 29/04/2020.
+//
+
+#ifndef DATASTRUCTURESWET1_TNODEIMP_H
+#define DATASTRUCTURESWET1_TNODEIMP_H
+
+
+#include "TNode.h"
+
+using std::make_shared;
+
+template<class Key, class Info>
+TNode<Key, Info>::TNode() = default;
+
+template<class Key, class Info>
+TNode<Key, Info>::~TNode() = default;
```

Comment:
for cases like those you can write it in the header file 🤔

Reply:
according to this it's better to implement in the body. As a good practice: https://stackoverflow.com/questions/43733938/default-c-definitions-in-body-or-header-files

Reply:
hmm... but i think that if we look at the header we want to know if it uses the default ctor aren't we?
if you decide not to change it just resolve